share/smack/lib/smack.c

```diff
@@ -2053,18 +2053,17 @@ void __SMACK_decls() {
   D("procedure {:inline 1} $$alloc(n: ref) returns (p: ref)\n"
     "modifies $Alloc, $CurrAddr;\n"
     "{\n"
-    "  p := $CurrAddr;\n"
-    "  havoc $CurrAddr;\n"
+    "  assume $sge.ref.bool(n, $0.ref);\n"
```

Comment:
Doesn’t this request address some recently-opened issue by @Guoanshisb ? If so the commit should probably reference the issue.